### DISCUSSION TITLE
Is there a way to silence log/debug output?

### DISCUSSION THREAD

**Discussion by oelna** [ggml-org/llama.cpp#1758] [Q&A] [answered]
When I run the program, there is a lot of output *other than the model response*, such as

```
main: build = 0 (unknown)
main: seed  = 1686229235
llama.cpp: loading model from /models/selfee-13b.ggmlv3.q4_0.bin
llama_model_load_internal: format     = ggjt v3 (latest)
llama_model_load_internal: n_vocab    = 32001
llama_model_load_internal: n_ctx      = 512
llama_model_load_internal: n_embd     = 5120
…
llama_print_timings: prompt eval time =  9715.31 ms /     5 tokens ( 1943.06 ms per token)
llama_print_timings:        eval time = 28065.27 ms /   337 runs   (   83.28 ms per token)
llama_print_timings:       total time = 38073.44 ms
```

Is there a way to prevent this output of log messages and stats, such as a verbosity level I can set as a parameter? Or a `--quiet` or `--silent` mode?

I would like to use the output in other contexts, such as shell scripts, but parsing out the log messages seems like the wrong way to go about it. I'm grateful for any tips or hints.

**Comment by j-f1** [comment, marked as answer]
If you redirect stderr you should only see the output (e.g. `./main 2>/dev/null`)

**Reply by oelna** [reply]
Amazingly simple solution. Thank you!

**Reply by waffletower** [reply]
`llama-cli` still litters the current directory with a `main.log` file while using this STDERR redirect solution. I prefer to use the `--log-disable` flag:
``` shell
llama-cli --log-disable --prompt "I have nothing to say and I am saying it."
```

**Reply by hasaranga** [reply]
what can I use on windows?

**Reply by rhatdan** [reply]
We are trying to build github.com/containers/ramalama which is wriapping llama-cli, and want to block the debugging noice, and are currently doing the 2> /dev/null thing, but this also blocks real errors from showing up.

When I tried the --log-disable, I see nothing almost like stdout and stderr are blocked, command just freezes.

```
llama-cli -m /home/dwalsh/.local/share/ramalama/models/ollama/tinyllama:latest --in-prefix '' --in-suffix '' -p 'You are a helpful assistant' -c 2048 -cnv --log-disable
```

But:

```
$ llama-cli -m /home/dwalsh/.local/share/ramalama/models/ollama/tinyllama:latest --in-prefix '' --in-suffix '' -p 'You are a helpful assistant' -c 2048 -cnv 2>/dev/null 
.....................................................................................
main: interactive mode on.

== Running in interactive mode. ==
 - Press Ctrl+C to interject at any time.
 - Press Return to return control to the AI.
 - To return control without starting a new line, end your input with '/'.
 - If you want to submit another line, end your input with '\'.

 You are a helpful assistant
> 
```

Issue we have is:
`llama-cli -m /home/dwalsh/.local/share/ramalama/models/ollama/missing --in-prefix '' --in-suffix '' -p 'You are a helpful assistant' -c 2048 -cnv 2>/dev/null`

Does not show the error:

```
llama_model_load: error loading model: llama_model_loader: failed to load model from /home/dwalsh/.local/share/ramalama/models/ollama/missing
```

**Reply by ericcurtin** [reply]
This was briefly discussed here when the behaviour changed https://github.com/ggerganov/llama.cpp/pull/9418

**Comment by missionsave** [comment]
c++:

`static void llama_log_callback_null(ggml_log_level level, const char * text, void * user_data) {
    (void) level;
    (void) text;
    (void) user_data;
}`

then call
`llama_log_set(llama_log_callback_null, NULL);`

**Reply by RhinoDevel** [reply]
Maybe an alternative: `common_log_pause(common_log_main());`

**Comment by missionsave** [comment]
llama_log_set([](ggml_log_level level, const char * text, void * user_data) {
        (void) level;
        (void) text;
        (void) user_data;
    }, NULL);
